Sources/ApolloCodegenTestSupport/MockNetworkSession.swift

```diff
@@ -0,0 +1,24 @@
+@testable import ApolloCodegenLib
+
+public final class MockNetworkSession: NetworkSession {
+  let statusCode: Int
+  let data: Data?
+  let error: Error?
+  let abandon: Bool
+
+  public init(statusCode: Int, data: Data? = nil, error: Error? = nil, abandon: Bool = false) {
```

Comment:
What is the `abandon` parameter here?

Reply:
`abandon` is used to mock a timeout error. Idea being to [exit early](https://github.com/apollographql/apollo-ios/pull/1935/files/6216e332256f63d2cf92a40da14480f9b25e8058#diff-aa32e38e100398072d34522c78690f7e3d530e6d3ad6ba6999a4507c1e553ddcR17) and never have the completion block called = response took longer than the timeout value used in the download configuration.